test/dshell/extra-files/cpp_header_gen/app.cpp

```diff
@@ -1,4 +1,9 @@
 // This should be solved by the header generator
+
+#define BEGIN_ENUM(name, upper, lower) enum name {
+#define ENUM_KEY(type, name, value, enumName, upper, lower, abbrev) upper##name = value,
+#define END_ENUM(name, upper, lower) };
```

Comment:
@ibuclaw DMD FE wants enum's like this right?

Reply:
Where the basetype isn't an int, a typedef is needed too.

Reply:
They emit a different macro which would be defined differently.
There is a few variations of the macro where C++ has hard language separations on how it is able to express enums.